### PR TITLE
[ADV-9623] Implement Ogury adapter

### PR DESCRIPTION
**Pull request by ghost**
# Summary

Implement the Ogury adapter (from the old MoPub custom events) with some refactoring.

### CODE REVIEW

File: Ogury/OguryAdapterConfiguration.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//
+//  Copyright © 2019 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryAdapterConfiguration.h]: 2019 -> 2021

File: Ogury/OguryAdapterConfiguration.m [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    return OguryConfigurationNetworkName;
+}
+
+- (NSString *)networkSdkVersion {
+    return [[OguryAds shared] sdkVersion];
+}
+
+#pragma mark - Methods
+
++ (void)updateInitializationParameters:(NSDictionary *)parameters {
```
Comment (cvlaminck) [inline Ogury/OguryAdapterConfiguration.m:45]: We cannot update the asset key in the SDK so I think we can keep this method empty.

File: Ogury/OguryAdapterConfiguration.m [hunk 3]
```diff
@@ -0,0 +1,80 @@
+//
+//  Copyright © 2019 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryAdapterConfiguration.m]: -> 2021

File: Ogury/OguryBannerCustomEvent.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
+//
+//  Copyright © 2020 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryBannerCustomEvent.h]: -> 2021

File: Ogury/OguryBannerCustomEvent.m [hunk 5]
```diff
@@ -0,0 +1,118 @@
+//
+//  Copyright © 2020 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryBannerCustomEvent.m]: -> 2021

File: Ogury/OguryBannerCustomEvent.m [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+- (void)oguryAdsBannerAdClosed:(OguryAdsBanner *)bannerAds {
+     [self.delegate inlineAdAdapterDidEndUserAction:self];
+}
+
+- (void)oguryAdsBannerAdDisplayed:(OguryAdsBanner *)bannerAds {
+     [self.delegate inlineAdAdapterDidTrackImpression:self];
+}
+
+- (void)oguryAdsBannerAdError:(OguryAdsErrorType)errorType forBanner:(OguryAdsBanner *)bannerAds {
+    NSError *error = [NSError errorWithDomain:kOguryErrorDomain code:MOPUBErrorNoInventory userInfo:@{NSLocalizedDescriptionKey: @"Failed to display Banner."}];
```
Comment (cvlaminck) [inline Ogury/OguryBannerCustomEvent.m]: You need to convert the Ogury code into MoPub code.

File: Ogury/OguryInterstitialCustomEvent.h [hunk 7]
```diff
@@ -0,0 +1,19 @@
+//
+//  Copyright © 2019 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryInterstitialCustomEvent.h]: -> 2021

File: Ogury/OguryInterstitialCustomEvent.m [hunk 8]
```diff
@@ -0,0 +1,100 @@
+//
+//  Copyright © 2019 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryInterstitialCustomEvent.m]: -> 2021

File: Ogury/OguryInterstitialCustomEvent.m [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+    [self.delegate fullscreenAdAdapterAdDidDismiss:self];
+}
+
+- (void)oguryAdsInterstitialAdDisplayed {
+    [self.delegate fullscreenAdAdapterAdDidAppear:self];
+    [self.delegate fullscreenAdAdapterDidTrackImpression:self];
+}
+
+- (void)oguryAdsInterstitialAdError:(OguryAdsErrorType)errorType {
+    NSError *error = [NSError errorWithDomain:kOguryErrorDomain code:MOPUBErrorNoInventory userInfo:@{NSLocalizedDescriptionKey: @"Failed to display Interstitial."}];
```
Comment (cvlaminck) [inline Ogury/OguryInterstitialCustomEvent.m]: Same as banner, we need to convert error code.

File: Ogury/OguryRewardedVideoCustomEvent.h [hunk 10]
```diff
@@ -0,0 +1,19 @@
+//
+//  Copyright © 2019 Ogury Ltd. All rights reserved.
```
Comment (cvlaminck) [inline Ogury/OguryRewardedVideoCustomEvent.h]: -> 2021

File: Ogury/OguryRewardedVideoCustomEvent.m [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+    [self.delegate fullscreenAdAdapterAdDidDismiss:self];
+}
+
+- (void)oguryAdsOptinVideoAdDisplayed {
+    [self.delegate fullscreenAdAdapterAdDidAppear:self];
+    [self.delegate fullscreenAdAdapterDidTrackImpression:self];
+}
+
+- (void)oguryAdsOptinVideoAdError:(OguryAdsErrorType)errorType {
+    NSError *error = [NSError errorWithDomain:kOguryErrorDomain code:MOPUBErrorNoInventory userInfo:@{NSLocalizedDescriptionKey: @"Failed to display RewardedVideo."}];
```
Comment (cvlaminck) [inline Ogury/OguryRewardedVideoCustomEvent.m]: Same as for interstitial.

File: Ogury/README.md [hunk 12]
```diff
@@ -0,0 +1,7 @@
+## Overview
+  * This folder contains mediation adapters used to mediate Ogury.
+  * To download and integrate the Ogury SDK, please check [this tutorial](https://ogury-ltd.gitbook.io/ios).
```
Comment (cvlaminck) [inline Ogury/README.md]: Use https://docs.ogury.co as top level domain for the doc.

File: Ogury/README.md [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
+  * For inquiries and support, please email [techops-supply@ogury.co](mailto:techops-supply@ogury.co).
+  
+## Adapter integration
+  * To integrate adapters, please visit our [integration tutorial](https://ogury-ltd.gitbook.io/ios).
```
Comment (cvlaminck) [inline Ogury/README.md]: Same as above.

**Review by cvlaminck — CHANGES_REQUESTED** [review]

File: Ogury/OguryAdapterConfiguration.h (hunk 1 above)
Comment (ghost) [reply Ogury/OguryAdapterConfiguration.h]: Done.

File: Ogury/OguryAdapterConfiguration.m (hunk 3 above)
Comment (ghost) [reply Ogury/OguryAdapterConfiguration.m]: Done.

File: Ogury/OguryBannerCustomEvent.h (hunk 4 above)
Comment (ghost) [reply Ogury/OguryBannerCustomEvent.h]: Done.

File: Ogury/OguryBannerCustomEvent.m (hunk 5 above)
Comment (ghost) [reply Ogury/OguryBannerCustomEvent.m]: Done.

File: Ogury/OguryInterstitialCustomEvent.h (hunk 7 above)
Comment (ghost) [reply Ogury/OguryInterstitialCustomEvent.h]: Done.

File: Ogury/OguryInterstitialCustomEvent.m (hunk 8 above)
Comment (ghost) [reply Ogury/OguryInterstitialCustomEvent.m]: Done.

File: Ogury/OguryRewardedVideoCustomEvent.h (hunk 10 above)
Comment (ghost) [reply Ogury/OguryRewardedVideoCustomEvent.h]: Done.

File: Ogury/README.md (hunk 12 above)
Comment (ghost) [reply Ogury/README.md]: Done.

File: Ogury/README.md (hunk 13 above)
Comment (ghost) [reply Ogury/README.md]: Done.

File: Ogury/OguryAdapterConfiguration.m (hunk 2 above)
Comment (ghost) [reply Ogury/OguryAdapterConfiguration.m:45]: Just checked the adaptor creation guidelines, I'll empty it.

File: Ogury/OguryBannerCustomEvent.m (hunk 6 above)
Comment (ghost) [reply Ogury/OguryBannerCustomEvent.m]: Done.

File: Ogury/OguryInterstitialCustomEvent.m (hunk 9 above)
Comment (ghost) [reply Ogury/OguryInterstitialCustomEvent.m]: Done.

File: Ogury/OguryRewardedVideoCustomEvent.m (hunk 11 above)
Comment (ghost) [reply Ogury/OguryRewardedVideoCustomEvent.m]: Done.

**Review by cvlaminck — APPROVED** [review]

**Pull request merged**